src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHandler.cs

```diff
@@ -2,29 +2,47 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Http.Diagnostics;
+using Microsoft.Extensions.Http.Resilience.Internal;
+using Microsoft.Shared.DiagnosticIds;
+using Microsoft.Shared.Diagnostics;
 using Polly;
 
-namespace Microsoft.Extensions.Http.Resilience.Internal;
+namespace Microsoft.Extensions.Http.Resilience;
 
 /// <summary>
 /// Base class for resilience handler, i.e. handlers that use resilience strategies  to send the requests.
 /// </summary>
-internal sealed class ResilienceHandler : DelegatingHandler
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Resilience, UrlFormat = DiagnosticIds.UrlFormat)]
+public class ResilienceHandler : DelegatingHandler
 {
     private readonly Func<HttpRequestMessage, ResiliencePipeline<HttpResponseMessage>> _pipelineProvider;
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ResilienceHandler"/> class.
+    /// </summary>
+    /// <param name="pipelineProvider">The pipeline provider that supplies pipelines in response to an http message.</param>
+    /// <exception cref="ArgumentNullException">If <paramref name="pipelineProvider"/> is <see langword="null"/>.</exception>
     public ResilienceHandler(Func<HttpRequestMessage, ResiliencePipeline<HttpResponseMessage>> pipelineProvider)
```

Comment:
How about alternative constructor for simple scenarios that just accepts `ResiliencePipeline`?

Reply:
Done.